std/format.d

```diff
@@ -3738,7 +3736,20 @@ if (is(T == class) && !is(T == enum))
         put(w, "null");
     else
     {
-        static if (hasToString!(T, Char) > 1 || (!isInputRange!T && !is(BuiltinTypeOf!T)))
+        static if ((is(T == immutable) || is(T == const) || is(T == shared)) && hasToString!(T, Char) == 0)
```

Comment:
I would prefer if we could use `== none` here.
See: https://github.com/dlang/phobos/pull/6209